### PR TITLE
Display ToS update notice

### PR DESCRIPTION
**Pull request by mmtr**
#### Changes proposed in this Pull Request

* Adds a selector that checks if user has accepted the current ToS.
* Adds an action for accepting current ToS.
* Displays a ToS update notice if user has not accepted the current ToS.
* Dispatches an action for accepting current ToS when user clicks on "Accept".

I used the same style guidelines we used on the GDPR banner (see p9mQOq-Hz-p2#comment-2068): 

Desktop | Medium | Mobile
--- | --- | ---
image | image | image

#### Testing instructions

* Comment out [this line](https://github.com/Automattic/wp-calypso/blob/a6c9835d0049ce05fe6793990cbbe159f424aa99/client/server/pages/index.js#L348) before starting Calypso locally, so the environment badge doesn't show up while testing.
* Make sure you don't have ToS marked as accepted. See D38262-code for instructions on how to reset ToS acceptance.
* Load any page in Calypso.
* Make sure the banner shows up and design looks ok in different viewports.
* Click on Accept.
* Make sure the banner is hidden and the ToS are accepted.
* Reload the current Calypso page.
* Make sure the banner doesn't show up anymore in any page.

Fixes https://github.com/Automattic/wp-calypso/issues/39046

### CODE REVIEW

**Comment by mmtr** [comment]
@danhauk / @sixhours I realize the colors of the banner are the same we used on the new upsell nudge component. Should we use different ones here?

File: client/blocks/legal-updates-banner/style.scss [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	display: flex;
+	flex-direction: column;
+
+	a {
+		color: var( --color-text-inverted );
+		text-decoration: underline;
+	}
+
+	.legal-updates-banner__content {
+		max-width: 800px;
```
Comment (kwight) [inline client/blocks/legal-updates-banner/style.scss]: Curious if this is arbitrary, or based on something?

File: client/blocks/legal-updates-banner/index.js [hunk 2]
```diff
@@ -1,22 +1,73 @@
...
+		props.requestLegalData();
 	}, [] );
 
+	if ( props.needsAcceptTos ) {
+		return (
+			<Card className="legal-updates-banner">
+				<div className="legal-updates-banner__content">
+					{ translate(
+						"We've updated our {{a}}Terms of Service{{/a}}. Please take a few moments to read them. By accepting, " +
+							'you agree to the updates Terms of Service.',
```
Comment (kwight) [inline client/blocks/legal-updates-banner/index.js]:
```suggestion
							'you agree to the new Terms of Service.',
```

File: client/state/selectors/should-display-tos-update-banner.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/**
+ * Indicates if a Terms or Service update banner should be displayed.
+ *
+ * @param state {object}
+ * @returns {boolean} Whether the ToS update banner should be displayed.
+ */
+export const shouldDisplayTosUpdateBanner = state => state?.legal?.tos?.displayPrompt;
```
Comment (kwight) [inline client/state/selectors/should-display-tos-update-banner.js]: 🎉

**Comment by kwight** [comment]
This all looks and works great for me (tested in RTL, too). Even if we have next steps for design, we can get this in and iterate from there 🎉

**Comment by kwight** [comment]
~~Oh wait – since this is using only `public-api` as the blog sticker check, it'll display to everyone as soon as it's merged?~~

Uh, no – because of [the flag](https://github.com/Automattic/wp-calypso/blob/master/client/layout/index.jsx#L201) that gates fetching the legal data. Carry on. 🙃

File: client/blocks/legal-updates-banner/style.scss (hunk 1 above)
Comment (mmtr) [reply client/blocks/legal-updates-banner/style.scss]: I used the same style guidelines we used on the GDPR banner (see p9mQOq-Hz-p2).

File: client/state/selectors/should-display-tos-update-banner.js (hunk 3 above)
Comment (mmtr) [reply client/state/selectors/should-display-tos-update-banner.js]: Having this centralized in the API has been tremendously helpful. Thanks for keeping our complexities wrapped in simplicity (I watched your talk! 😃).

File: client/blocks/legal-updates-banner/style.scss (hunk 1 above)
Comment (mmtr) [reply client/blocks/legal-updates-banner/style.scss]: image

**Comment by donalirl** [comment]
Maybe this has been caught somewhere already, but I figured it best to say something than not say something :) 

There is a typo in the Calypso Footer Banner. The second `updates` should say `updated`:

>you agree to the updates Terms of Service.

to

>you agree to the updated Terms of Service.

**Comment by mmtr** [comment]
Thanks for checking @donalirl! We finally used a different message but forgot to update the screenshots 🙂 

https://github.com/Automattic/wp-calypso/blob/410d381a93d263ec69698cc9ab3f2bde542b697e/client/blocks/legal-updates-banner/index.js#L34-L35

**Pull request merged**